actions/apps.ts

```diff
@@ -56,22 +118,83 @@ export function doAppCall<Res=unknown>(call: AppCallRequest, type: AppCallType,
                     });
                     return {error: makeCallErrorResponse(errMsg)};
                 }
+                if (shouldOpenInNewTab(res.navigate_to_url, getSiteURL())) {
+                    window.open(res.navigate_to_url);
+                    return {data: res};
+                }
+                browserHistory.push(res.navigate_to_url.slice(getSiteURL().length));
```

Comment:
What if the url is s relative path? I'm also concerned about using a site URL with a subpath cc @DHaussermann

```suggestion
                let navigateURL = res.navigate_to_url;
                if (navigateURL.startsWith(getSiteURL())) {
                    navigateURL = navigateURL.slice(getSiteURL().length);
                }
                browserHistory.push(navigateURL);
```

Reply:
@mickmister @hanzei I've been thinking about this, is there a reason to even allow relative URLs here? Even if we want to allow them as apps return values, we can normalize in the server, and consider them valid here. Also need to check the mobile handling.

Reply:
> Even if we want to allow them as apps return values, we can normalize in the server, and consider them valid here

@levb Makes sense to me